pcdsdevices/sxr_test_absorber.py

```diff
@@ -1,39 +1,51 @@
 """
 Module for the SXR Test Absorbers.
 """
-from lightpath import LightpathState
 from ophyd.device import Component as Cpt
 
 from .epics_motor import BeckhoffAxisNoOffset
-from .interface import BaseInterface, LightpathMixin
+from .inout import TwinCATInOutPositioner
+from .interface import BaseInterface, LightpathInOutCptMixin
+from .signal import PytmcSignal
 
 
-class SxrTestAbsorber(BaseInterface, LightpathMixin):
+class ST3K4AutoError(RuntimeError):
+    ...
+
+
+class SxrTestAbsorberStates(TwinCATInOutPositioner):
+    """
+    SXR Test absorber states
+
+    This has some automation where it tracks the position of ST3K4.
+    Accordingly, it has a configuration parameter for enabling/disabling this.
+    """
+    st3k4_auto = Cpt(PytmcSignal, ':ST3K4_AUTO', io='io', kind='config')
+
+    def check_value(self, pos):
+        rval = super().check_value(pos)
+        if self.st3k4_auto.get():
+            raise ST3K4AutoError(
+                "ST1K4 must follow ST3K4. Move rejected."
```

Comment:
Nice! Without this, I could see the "ST1K4 doesn't move!" Jira issues coming in as soon as this is deployed...